addons/sale_timesheet/models/account.py

```diff
@@ -63,7 +63,7 @@ def _compute_allowed_so_line_ids(self):
     @api.depends('project_id.partner_id.commercial_partner_id', 'task_id.partner_id.commercial_partner_id')
     def _compute_commercial_partner(self):
         for timesheet in self:
-            timesheet.commercial_partner_id = timesheet.task_id.partner_id.commercial_partner_id or timesheet.project_id.partner_id.commercial_partner_id
+            timesheet.commercial_partner_id = timesheet.task_id.sudo().partner_id.commercial_partner_id or timesheet.project_id.sudo().partner_id.commercial_partner_id
```

Comment:
is it really needed to add a sudo in `timesheet.task_id.partner_id.commercial_partner_id`? Normally, should be able to access to the task, no?